src/utils.zig

```diff
@@ -1024,6 +1030,46 @@ pub const ThreadPool = struct { // MARK: ThreadPool
 		isStillNeeded: *const fn(*anyopaque, milliTime: i64) bool,
 		run: *const fn(*anyopaque) void,
 		clean: *const fn(*anyopaque) void,
+		taskType: TaskType = .misc,
+	};
+	pub const Performance = struct {
+		mutex: std.Thread.Mutex = .{},
+		tasks: [taskTypes]u32,
+		utime: [taskTypes]i64,
+
+		fn add(self: *Performance, task: TaskType, time: i64) void {
+			self.mutex.lock();
+			defer self.mutex.unlock();
+			const i = @intFromEnum(task);
+			self.tasks[i] += 1;
+			self.utime[i] += time;
+		}
+
+		pub fn clear(self: *Performance) void {
+			self.mutex.lock();
+			defer self.mutex.unlock();
+			for(0..taskTypes) |i| {
+				self.tasks[i] = 0;
+				self.utime[i] = 0;
+			}
+		}
+
+		fn init(allocator: NeverFailingAllocator) *Performance {
+			const self = allocator.create(Performance);
+			self.clear();
+			return self;
+		}
+
+		pub fn read(self: *Performance) Performance {
+			var copy = Performance{.tasks = undefined, .utime = undefined};
+			self.mutex.lock();
+			defer self.mutex.unlock();
+			for(0..taskTypes) |i| {
+				copy.tasks[i] = self.tasks[i];
+				copy.utime[i] = self.utime[i];
+			}
+			return copy;
```

Comment:
Why not just `return self.*;`? That should automatically copy the entire thing.

Reply:
Oh, thanks.